storage/codex.js

```diff
@@ -0,0 +1,57 @@
+import { BaseStorageProvider } from "./base.js";
+import { Codex } from "@codex-storage/sdk-js";
+import { NodeUploadStategy } from "@codex-storage/sdk-js/node";
+
+class CodexStorageProvider extends BaseStorageProvider {
+    constructor({ codexURI }) {
+        super();
+        if (!codexURI) {
+            throw new Error("Codex URI is required");
+        }
+        this.codexURI = codexURI;
+        const client = new Codex(codexURI);
+        this.client = client;
+    }
+
+    async upload(fileName, content) {
+        const file = new File([content], fileName, { type: "text/plain" });
+
+        try {
+            const strategy = new NodeUploadStategy([content], fileName, { type: "text/plain" })
+    
+            const resultPromise = this.client.data.upload(strategy);
```

Comment:
I was thinking:
Uploading a file and returning the CID is good for proving interoperability, but it's not really how to store something in the network. For that, a few more steps are needed:
- Upload file, get CID
- Use CID + purchase parameters to create storage request
- Wait till request is started
- From the request info, get the encoded CID and return that one
I completely understand that maybe this isn't the time and place to implement all that.